simulation/simulate.py

```diff
@@ -53,13 +52,36 @@ def simulation_loop(self):
     i = 0
     while i < update_num:
       # Do autopilot controls
+
+      # Occasionally, airsim lags behind jsbsim, causing aircraft to spawn inside obstacles.
+      # Checking this and reinitializing helps to alleviate these cases.
+      if i == 0:
+        pose = self.sim.client.simGetVehiclePose()
+
+        # Experimentally determined, in UE4 coordinate system
+        ic_position = np.array([0, 0, -1.3411200046539307])
+        current_position = np.array([pose.position.x_val, pose.position.y_val, pose.position.z_val])
+        retry_period = 10
+        retry_counter = 0
+
+        while (np.abs(ic_position - current_position) > np.finfo(float).eps).all():
+          if retry_counter % retry_period == 0:
+            self.sim.reinitialize()
+          retry_counter += 1
+
+          pose = self.sim.client.simGetVehiclePose()
+          current_position = np.array([pose.position.x_val, pose.position.y_val, pose.position.z_val])
+          
```

Comment:
Is this something we can call before the beginning of the loop for clarity since it only happens when i = 0? Also, just double checking that the `ic_position` would need to be updated if we changed the initial conditions right?